File: conclave-init/template/client/src/main/java/com/r3/conclave/template/client/TemplateEnclaveClient.java
package com.r3.conclave.template.client;

import com.r3.conclave.client.EnclaveClient;
import com.r3.conclave.client.web.WebEnclaveTransport;
import com.r3.conclave.common.EnclaveConstraint;
import com.r3.conclave.common.InvalidEnclaveException;
import com.r3.conclave.mail.EnclaveMail;

import java.io.IOException;
import java.nio.charset.StandardCharsets;

class TemplateEnclaveClient {
    public static void main(String[] args) throws IOException, InvalidEnclaveException {
        if (args.length != 1) {
            System.out.println("Usage: java -jar client.jar ENCLAVE_CONSTRAINT");
            System.exit(1);
        }

        EnclaveConstraint constraint = EnclaveConstraint.parse(args[0]);

        // The URL has been hardcoded to localhost:8080, but this will need to be parametrised in a real client.
        try (WebEnclaveTransport transport = new WebEnclaveTransport("http://localhost:8080");
             EnclaveClient client = new EnclaveClient(constraint))
        {
            client.start(transport);
            byte[] requestMailBody = "abc".getBytes(StandardCharsets.UTF_8);
            EnclaveMail responseMail = client.sendMail(requestMailBody);
            String responseString = (responseMail != null) ? new String(responseMail.getBodyAsBytes()) : null;
            System.out.println("Enclave returned " + responseString);
        }
    }
}

File: conclave-init/template/client/src/main/kotlin/com/r3/conclave/template/client/TemplateEnclaveClient.kt
package com.r3.conclave.template.client

import com.r3.conclave.client.EnclaveClient
import com.r3.conclave.client.web.WebEnclaveTransport
import com.r3.conclave.common.EnclaveConstraint
import kotlin.system.exitProcess

object TemplateEnclaveClient {
    @JvmStatic
    fun main(args: Array<String>) {
        if (args.size != 1) {
            println("Usage: client/bin/client ENCLAVE_CONSTRAINT")
            exitProcess(1)
        }
        val constraint = EnclaveConstraint.parse(args.single())

        WebEnclaveTransport("http://localhost:8080").use { transport ->
            EnclaveClient(constraint).use { client ->
                client.start(transport)
                val responseMail = client.sendMail("abc".toByteArray())
                val responseString = responseMail?.let { String(it.bodyAsBytes) }
                println("Enclave returned '$responseString'")
            }
        }
    }
}